### PR TITLE
EZP-20723: linkcheck cronjob: some urls marked invalid due to HEAD request

### PR DESCRIPTION
**Pull request by joaoinacio**
JIRA: https://jira.ez.no/browse/EZP-20723

This uses CURLOPT_CUSTOMREQUEST ( GET ) together with CURLOPT_NOBODY options to mimic 'HEAD' requests when using curl.

In addition, also get result data from curl as string, instead of capturing the output buffer.

### CODE REVIEW

**Comment by andrerom** [comment]
+1 on this approach, review ping @glye

File: lib/ezutils/classes/ezhttptool.php [hunk 1]
```diff
@@ -768,32 +776,28 @@ static function getDataByURL( $url, $justCheckURL = false, $userAgent = false )
...
                 {
-                    curl_close( $ch );
-                    return false;
+                    return true;
+                }
+                if ( !$forceGetRequest )
+                {
+                  // if HEAD request failed with anything other than a 404, repeating using a GET request
+                  $httpCode = curl_getinfo( $ch, CURLINFO_HTTP_CODE );
+                  if ( $httpCode != 404 )
```
Comment (yannickroger) [inline lib/ezutils/classes/ezhttptool.php]: missing `{}`

**Comment by yannickroger** [comment]
+1

**Comment by glye** [comment]
+1 (sorry for delay, I don't get notification from github when I'm mentioned/pinged here)

**Comment by patrickallaert** [comment]
+1 on the concept (falling back on `GET` after `HEAD` not behaving correctly).

PS: I haven't reviewed the code.

**Pull request merged**